common/buildcraft/core/ItemMapLocation.java

```diff
@@ -93,64 +96,102 @@ public void addInformation(ItemStack stack, EntityPlayer player, @SuppressWarnin
+                    cpt.setInteger("yMin", areaTile.min().getY());
+                    cpt.setInteger("zMin", areaTile.min().getZ());
+                    cpt.setInteger("xMax", areaTile.max().getX());
+                    cpt.setInteger("yMax", areaTile.max().getY());
+                    cpt.setInteger("zMax", areaTile.max().getZ());
+
+                } else {
+                    MapLocationType.SPOT.setToStack(toChange);
+
+                    cpt.setByte("side", (byte) movingobjectposition.sideHit.getIndex());
+                    cpt.setInteger("x", pos.getX());
+                    cpt.setInteger("y", pos.getY());
+                    cpt.setInteger("z", pos.getZ());
+                }
+                if (toChange != stack && !player.inventory.addItemStackToInventory(toChange)) {
+                    player.dropItem(toChange, false, true);
+                }
             }
+        } else if (player.isSneaking()) {
+            clearMapLocation(stack);
+        }
+        return stack;
+    }
 
-            cpt.setTag("path", pathNBT);
-        } else if (tile instanceof IAreaProvider) {
-            MapLocationType.AREA.setToStack(modified);
-
-            IAreaProvider areaTile = (IAreaProvider) tile;
-
-            cpt.setInteger("xMin", areaTile.min().getX());
-            cpt.setInteger("yMin", areaTile.min().getY());
-            cpt.setInteger("zMin", areaTile.min().getZ());
-            cpt.setInteger("xMax", areaTile.max().getX());
-            cpt.setInteger("yMax", areaTile.max().getY());
-            cpt.setInteger("zMax", areaTile.max().getZ());
-
-        } else {
-            MapLocationType.SPOT.setToStack(modified);
-
-            cpt.setByte("side", (byte) side.getIndex());
-            cpt.setInteger("x", pos.getX());
-            cpt.setInteger("y", pos.getY());
-            cpt.setInteger("z", pos.getZ());
+    protected MovingObjectPosition getMovingObjectPositionFromPlayer(World worldIn, EntityPlayer playerIn) {
+        float f = playerIn.rotationPitch;
+        float f1 = playerIn.rotationYaw;
+        double d0 = playerIn.posX;
+        double d1 = playerIn.posY + (double) playerIn.getEyeHeight();
+        double d2 = playerIn.posZ;
+        Vec3 vec3 = new Vec3(d0, d1, d2);
+        float f2 = MathHelper.cos(-f1 * 0.017453292F - (float) Math.PI);
+        float f3 = MathHelper.sin(-f1 * 0.017453292F - (float) Math.PI);
+        float f4 = -MathHelper.cos(-f * 0.017453292F);
+        float f5 = MathHelper.sin(-f * 0.017453292F);
+        float f6 = f3 * f4;
+        float f7 = f2 * f4;
+        double d3 = 5.0D;
+        if (playerIn instanceof net.minecraft.entity.player.EntityPlayerMP) {
+            d3 = ((net.minecraft.entity.player.EntityPlayerMP) playerIn).theItemInWorldManager.getBlockReachDistance();
         }
+        Vec3 vec31 = vec3.addVector((double) f6 * d3, (double) f5 * d3, (double) f7 * d3);
+        return worldIn.rayTraceBlocks(vec3, vec31, false, false, false);
+    }
```

Comment:
Why is this necessary? Is this copied from somewhere in minecraft or did you just write all of this?